src/platform/packages/shared/kbn-unified-data-table/src/components/data_table.tsx

```diff
@@ -1475,6 +1475,12 @@ const componentStyles = {
       '.unifiedDataTable__cell--expanded': {
         backgroundColor: euiTheme.colors.backgroundBaseInteractiveSelect,
       },
+      '.euiDataGridRowCell__content--defaultHeight > div': {
+        maxInlineSize: '100%',
+        overflow: 'hidden',
+        textOverflow: 'ellipsis',
+        whiteSpace: 'nowrap',
+      },
```

Comment:
It seems like the root cause of the issue may be that we wrap cell contents in a `div` which defaults to `display: block`, preventing the EUI styles from being inherited correctly. Instead of redefining the EUI styles here, what if we just change the wrapping element to a `span` in `InTableSearchHighlightsWrapper` so the styles are inherited correctly?
https://github.com/elastic/kibana/blob/faf0c2852f9be73aa72572cfc92c635a9650a706/src/platform/packages/shared/kbn-data-grid-in-table-search/src/in_table_search_highlights_wrapper.tsx#L64

I gave it a quick shot locally and it seemed to work as expected.

Reply:
+1 on this 😄

Reply:
That's a very good idea, I focused on finding the culprit div so hard that I didn't see the simplest solution 😁 Thanks, will apply the changes today